services/shhext/chat/x3dh.go

```diff
@@ -43,28 +44,36 @@ func FromBase64(str string) (*Bundle, error) {
 	return bundle, nil
 }
 
-func buildSignatureMaterial(signedPreKeys *map[string]*SignedPreKey) []byte {
+func buildSignatureMaterial(bundle *Bundle) []byte {
+	signedPreKeys := bundle.GetSignedPreKeys()
+	timestamp := bundle.GetTimestamp()
 	var keys []string
-	for k := range *signedPreKeys {
+
+	for k := range signedPreKeys {
 		keys = append(keys, k)
 	}
 	var signatureMaterial []byte
 
 	sort.Strings(keys)
 
 	for _, installationID := range keys {
-		signedPreKey := (*signedPreKeys)[installationID]
+		signedPreKey := signedPreKeys[installationID]
 		signatureMaterial = append(signatureMaterial, []byte(installationID)...)
 		signatureMaterial = append(signatureMaterial, signedPreKey.SignedPreKey...)
 		signatureMaterial = append(signatureMaterial, []byte(fmt.Sprint(signedPreKey.Version))...)
+		// We don't use timestamp in the signature if it's 0, for backward compatibility
+	}
+
+	if timestamp != 0 {
+		signatureMaterial = append(signatureMaterial, []byte(fmt.Sprint(timestamp))...)
```

Comment:
Maybe `strconv.Itoa()` instead of `fmt`? Also, I guess it's ok in this case but more proper conversion of integers to bytes would be with `encoding/binary` package.